### PR TITLE
Saving a syntetic indicator for the ground motion fields and optimized the GmfGetter

### PR DESCRIPTION
**Pull request by micheles**
For the purpose of studying the seed-dependency it is important to consider statistics quantities. For this purpose I am storing the mean mean ground motion (mean on all sites and mean on all events) for each realization. This is for internal use only and not exported.
Here is how one could use the `gmv` indicator to study the seed dependency of our demo:

```
$ oq run demos/hazard/EventBasedPSHA/job.ini --param ses_seed=10 && oq show gmdata
PGA:float32,events:uint32,nbytes:uint32
3.371736E-02,176,2288


$ oq run demos/hazard/EventBasedPSHA/job.ini --param ses_seed=20 && oq show gmdata
PGA:float32,events:uint32,nbytes:uint32
3.230543E-02,176,2288
```

Even if there are only 176 events the indicator is rather stable.

PS: in this PR there is also an optimization of the hazard getters. Now the `GmfGetter` for event based hazard with no hazard curves is a lot faster than before and consumes a lot less memory. Here are some numbers for a computation with 1792 sites, 73807 events:

```
================================ ======== ========= ======
master                           time_sec memory_mb counts
================================ ======== ========= ======
total compute_gmfs_and_curves    692      807       20    
complete runtime                 178      2,358     1     
total compute_ruptures           68       23        13    
making contexts                  56       10        20    
filtering ruptures               46       0.0       9,040 
managing sources                 2.307    1.820     1     
saving gmfs                      2.052    37        20    

================================ ======== ========= ======
this branch                       time_sec memory_mb counts
================================ ======== ========= ======
total compute_gmfs_and_curves    415      313       20    
complete runtime                 118      2,357     1     
total compute_ruptures           68       22        13    
making contexts                  56       34        20    
filtering ruptures               46       0.0       9,040 
saving gmfs                      2.918    78        20    
managing sources                 2.505    0.547     1     
```

### CODE REVIEW

File: openquake/risklib/riskinput.py [hunk 1]
```diff
@@ -393,15 +396,15 @@ def __iter__(self):
     def __len__(self):
         return len(self._riskmodels)
 
-    def build_input(self, rlzs, hazards_by_site, assetcol, eps_dict):
+    def build_input(self, imts, rlzs, hazards_by_site, assetcol, eps_dict):
         """
         :param rlzs: a list of realizations
```
Comment (ptormene) [inline openquake/risklib/riskinput.py]: Please add the description of `imts` to the docstring

File: openquake/risklib/riskinput.py [hunk 2]
```diff
@@ -564,7 +572,8 @@ class RiskInput(object):
     :param assets_by_site: array of assets, one per site
     :param eps_dict: dictionary of epsilons
     """
-    def __init__(self, rlzs, hazard_by_site, assets_by_site, eps_dict):
+    def __init__(self, imts, rlzs, hazard_by_site, assets_by_site, eps_dict):
```
Comment (ptormene) [inline openquake/risklib/riskinput.py]: Please update the docstring accordingly

File: openquake/risklib/riskinput.py [hunk 3]
```diff
@@ -468,14 +471,19 @@ def __repr__(self):
 
 class PoeGetter(object):
     """
-    Callable yielding dictionaries {imt: curve} when called on a realization.
+    Callable yielding a matrix of poes when called on a realization.
```
Comment (ptormene) [inline openquake/risklib/riskinput.py]: Please add the description of parameters if needed

File: openquake/calculators/event_based_risk.py [hunk 4]
```diff
@@ -546,15 +547,14 @@ def save_losses(self, agglosses, asslosses, offset):
 
     def post_execute(self, num_events):
         """
-        Save an array of losses by taxonomy of shape (T, L, R).
+        Save risk data
         """
         event_based.EventBasedRuptureCalculator.__dict__['post_execute'](
             self, num_events)
-        if self.gmfbytes == 0:
+        gmv = sum(gm[:-2].sum() for gm in self.gmdata.values())
```
Comment (ptormene) [inline openquake/calculators/event_based_risk.py]: Please add a comment also here about indices

File: openquake/risklib/riskinput.py (hunk 1 above)
Comment (micheles) [inline openquake/risklib/riskinput.py]: Done

File: openquake/risklib/riskinput.py (hunk 2 above)
Comment (micheles) [inline openquake/risklib/riskinput.py]: Done

File: openquake/calculators/event_based_risk.py (hunk 4 above)
Comment (micheles) [inline openquake/calculators/event_based_risk.py]: Done

**Pull request merged**